spec/lib/distributed_mutex_spec.rb

```diff
@@ -40,26 +40,19 @@
     expect(Time.now.to_i).to be <= start + 1
   end
 
-  #  expected: 1574200319
-  #       got: 1574200320
-  #
-  #  (compared using ==)
-  # ./spec/components/distributed_mutex_spec.rb:60:in `block (3 levels) in <main>'
-  # ./lib/distributed_mutex.rb:33:in `block in synchronize'
-  xit 'allows the validity of the lock to be configured' do
+  it "allows the validity of the lock to be configured" do
     freeze_time
-
     mutex = DistributedMutex.new(key, validity: 2)
 
     mutex.synchronize do
-      expect(Discourse.redis.ttl(key)).to eq(2)
+      expect(Discourse.redis.ttl(key)).to eq(3)
```

Comment:
This feels kinda risky... `freeze_time` is not gonna affect Redis's internal clock. So if it takes a little longer for this line to be run, the ttl could be lower? Maybe that's why this test was skipped?

Maybe we should be more generous, like `.to be <= 3` ? 🤔

(and similarly on the other ttl check a few lines lower)

Reply:
Updated the spec. What do you think? 🙂

Reply:
Looks great to me 🚀